fujiji-server/src/controllers/comment.js

```diff
@@ -126,4 +127,47 @@ async function editComment(req, res, next) {
   }
 }
 
-module.exports = { postComment, getListingComments, editComment };
+// PURPOSE: implement the del comment endpoint
+async function deleteComment(req, res, next) {
+  const { comment_id: commentID } = req.params;
+  const { id } = req.locals.userData;
+  const userID = parseInt(id, 10);
+  const intCommentId = parseInt(commentID, 10);
+
+  const existingComment = await getCommentById(intCommentId);
+
+  if (!existingComment) {
+    return next(
+      new CommentNotFoundError(`comment with id:${commentID} is not found`),
+    );
+  }
+
+  const listingID = existingComment.listing_id;
+  const listing = await getListingById(parseInt(listingID, 10));
+
+  if (!listing) {
+    return next(
+      new ListingNotFound(`listing with id:${listingID} is not found`),
+    );
+  }
+
+  const isCommenter = userID === parseInt(existingComment.user_id, 10) ? 1 : 0;
+
+  const isAuthor = userID === parseInt(listing.user_id, 10) ? 1 : 0;
+
+  if (!isCommenter && !isAuthor) {
+    return res.status(400).json({ error: 'User does not own the comment nor is the seller of the listing.' });
+  }
```

Comment:
Can we change this condition to `!isCommenter`? We don't need the isAuthor check. If the person is the commenter then they can delete the comment.

Reply:
I have removed seller from deleting the comment.